### ISSUE TITLE
Deactivating portrait mode does nothing on iOS

### ISSUE THREAD

**Issue by C29H25N3O5** [26F-Studio/Techmino#847]
Even when I had portrait mode deactivated in settings the game still starts with portrait mode. The only way to enter landscape mode is to open the game with the device in landscape mode already.

**Comment by bytemain** [comment]
Actually, I found it can only launch in portrait mode on iOS 16 when you holding the phone vertically.

but in the same scene, it will launch in landscape mode on iOS 15.

<img width="1858" alt="CleanShot 2023-02-19 at 20 22 05@2x" src="https://user-images.githubusercontent.com/13938334/219947656-db6b1a4e-8963-43d5-a32d-76baae64502b.png">

But I tried search things on repos of SDL2 and love2d, didn't found anything

maybe this commit fixed this issue: https://github.com/libsdl-org/SDL/commit/70656b133c89c614422130b6c7e0a52cdccc6d4f#diff-de1e3f37ab58f8229cf963eff7070fde81645400705c9f34aff43d9a84af9418

**Comment by bytemain** [comment]
When I tried to use a simple demo to reproduce, it just behaves right both on iOS 15 and iOS 16.

![CleanShot 2023-02-19 at 20 41 32@2x](https://user-images.githubusercontent.com/13938334/219948621-ab566ead-76bc-4bbc-8730-dc65495ff4e1.png)

ios related source code are the same.

**Comment by ParticleG** [comment]
> 

What's your love2d revision? Techmino iOS uses [c35356c](https://github.com/love-actions/love-actions-ios/blob/5828881a1f03277e78cb856376ea533bb9689dbc/action.yml#L36), see [here](https://github.com/love2d/love/tree/c35356c841976eb6f370347b81eec845d5520338)

**Comment by ParticleG** [comment]
> Actually, I found it can only launch in portrait mode on iOS 16 when you holding the phone vertically.
> 
> but in the same scene, it will launch in landscape mode on iOS 15.
> 
> <img alt="CleanShot 2023-02-19 at 20 22 05@2x" width="1858" src="https://user-images.githubusercontent.com/13938334/219947656-db6b1a4e-8963-43d5-a32d-76baae64502b.png">
> 
> But I tried search things on repos of SDL2 and love2d, didn't found anything
> 
> maybe this commit fixed this issue: [libsdl-org/SDL@70656b1#diff-de1e3f37ab58f8229cf963eff7070fde81645400705c9f34aff43d9a84af9418](https://github.com/libsdl-org/SDL/commit/70656b133c89c614422130b6c7e0a52cdccc6d4f#diff-de1e3f37ab58f8229cf963eff7070fde81645400705c9f34aff43d9a84af9418)

What's your love2d revision? Techmino iOS uses [c35356c](https://github.com/love-actions/love-actions-ios/blob/5828881a1f03277e78cb856376ea533bb9689dbc/action.yml#L36), see [here](https://github.com/love2d/love/tree/c35356c841976eb6f370347b81eec845d5520338)

**Comment by bytemain** [comment]
not relevant with the version of love2d I'm using,  if you download from the App Store then you will meet the same problem.

**Comment by ParticleG** [comment]
> not relevant with the version of love2d I'm using, if you download from the App Store then you will meet the same problem.

It seems this issue [love-android #196](https://github.com/love2d/love-android/issues/196) happens on iOS16 again. You can try these configs in love.conf(): 
```lua
t.window.width,t.window.height=1280,720
t.window.fullscreen=true
t.window.resizeable=false
```

**Comment by ParticleG** [comment]
This should be temporarily fixed by removing the `W.fullscreen` config in conf.lua  #862

**Comment by bytemain** [comment]
<img width="1872" alt="CleanShot 2023-02-21 at 15 11 29@2x" src="https://user-images.githubusercontent.com/13938334/220273299-60718197-1a03-4c68-a5b0-62df23b4db67.png">

tested `ci-ios-test` branch, works well.

**Comment by ParticleG** [comment]
> <img alt="CleanShot 2023-02-21 at 15 11 29@2x" width="1872" src="https://user-images.githubusercontent.com/13938334/220273299-60718197-1a03-4c68-a5b0-62df23b4db67.png">
> 
> tested `ci-ios-test` branch, works well.

Could you test these configs on your simple demo? We currently don't have a mac to simulate iOS 15 devices.

> > not relevant with the version of love2d I'm using, if you download from the App Store then you will meet the same problem.
> 
> It seems this issue [love-android #196](https://github.com/love2d/love-android/issues/196) happens on iOS16 again. You can try these configs in love.conf():
> 
> ```lua
> t.window.width,t.window.height=1280,720
> t.window.fullscreen=true
> t.window.resizeable=false
> ```

**Comment by ParticleG** [comment]
These configs should make your demo launch in landscape mode on iOS 15, but in portrait mode on iOS 16

**Comment by bytemain** [comment]
Ok, I will try it now.

**Comment by bytemain** [comment]
```lua
		t.window.width, t.window.height = 1280, 720
		t.window.fullscreen = true
		t.window.resizeable = false
```

caused ios 15 launch in portrait mode too.

**Comment by bytemain** [comment]
<img width="1846" alt="CleanShot 2023-02-21 at 15 47 45@2x" src="https://user-images.githubusercontent.com/13938334/220280838-40624614-05b0-4bd4-bdc1-4c50fdb65e5f.png">

both launched in landscape mode

<img width="1219" alt="CleanShot 2023-02-21 at 15 48 36@2x" src="https://user-images.githubusercontent.com/13938334/220281025-8e3f119b-401b-4dda-9837-574b7af5ab28.png">

**Comment by ParticleG** [comment]
That's weird...

**Comment by bytemain** [comment]
But ci-ios-test branch actually worked, and works on my real device(ios 16.3) too.

**Comment by bytemain** [comment]
https://gist.github.com/bytemain/0e4e93e4cc195c9c744518583005fc4b

demo which I used for testing

**Comment by ParticleG** [comment]
![image](https://user-images.githubusercontent.com/38580644/220292989-e51228f3-eeb7-4445-b5dc-0a328ba7f7f2.png)
In Techmino's `conf.lua`, this would cause portrait on iOS 16. I'll try your `main.lua`

**Comment by ParticleG** [comment]
> https://gist.github.com/bytemain/0e4e93e4cc195c9c744518583005fc4b
> 
> demo which I used for testing

You haven't tried using `conf.lua`? Wiki said it woudn't work  if defined in `main.lua`.

![image](https://user-images.githubusercontent.com/38580644/220299526-6fa00e81-b97e-44ca-93c5-b7b947cb5d2b.png)

Try newest code in `ci-ios-test`, it would cause demo launch in portrait mode on iOS 16.
It uses a `conf.lua` with demo `main.lua`

**Comment by bytemain** [comment]
oh I forget it, let me try again

**Comment by ParticleG** [comment]
![0605DCFD-473A-48E2-96B7-9148A13B2F83](https://user-images.githubusercontent.com/38580644/220302058-d0f037a5-a59a-4e79-9708-9604f29b69f4.png)
This is newest build on `ci-ios-test`

**Comment by bytemain** [comment]
<img width="1122" alt="CleanShot 2023-02-21 at 17 19 58@2x" src="https://user-images.githubusercontent.com/13938334/220302350-82fe1185-c9e9-4743-a239-8ba2eca911f5.png">


You are right.

**Comment by ParticleG** [comment]
Okay, fullscreen config affects orientation again. so this should be an upstream issue of Love, I'll report an issue to them

**Comment by RubixYouTube** [comment]
`da code
<p style="text-align:center"><strong><em><span style="font-family:Courier New,Courier,monospace"><span style="font-size:36px">Frequency Incremental</span></span></em></strong></p>

<p style="text-align:center"><strong><em><span style="font-family:Courier New,Courier,monospace"><span style="font-size:20px">Update!!!</span></span></em></strong></p>

<p><strong><em><span style="font-family:Courier New,Courier,monospace"><span style="font-size:28px">Hz Meter:&nbsp;</span></span><span style="font-size:28px"><span style="font-family:Times New Roman,Times,serif">0 Hz</span></span></em></strong></p>

<p><strong><em><span style="font-size:28px"><span style="font-family:Times New Roman,Times,serif"><input name="Gain1" type="button" value="Get 1 Hz" />&nbsp;<input name="Gain2" type="reset" value="RESET." /></span></span></em></strong></p>

<p style="text-align:center"><strong><em><span style="font-family:Times New Roman,Times,serif"><span style="font-size:72px"><a href="https://github.com/RubixYoutube">Menu.</a></span></span></em></strong></p>

<p style="text-align:center"><strong><em><span style="font-family:Times New Roman,Times,serif"><span style="font-size:36px">Notice:&nbsp;</span></span><span style="font-size:36px"><span style="font-family:Georgia,serif">It Got Improved.&nbsp;</span><span style="font-family:Courier New,Courier,monospace">So Don&#39;t Mess The Code.</span></span></em></strong></p>
`
